### ISSUE TITLE
php grammar-v4 not working

### ISSUE THREAD

**Issue by Novitoll** [renatahodovan/grammarinator#35]
I'm following the instructions in [README](grammarinator-process PhpLexer.g4 PhpParser.g4 -o out)

Download PhpLexer.g4 and PhpParser.g4 from [grammars-v4](https://github.com/antlr/grammars-v4/tree/master/php), then

```bash
$ grammarinator-process PhpLexer.g4 PhpParser.g4 -o out
$ tree out
├── PhpUnlexer.py
└── PhpUnparser.py
```

I've also downloaded grammars-v4 [PhpLexerBase.py](https://github.com/antlr/grammars-v4/blob/master/php/Python/PhpLexerBase.py) and placed it to `out/`.

Now when I try to generate:

```
$ grammarinator-generate -p out/PhpUnparser.py -l out/PhpUnlexer.py -o generated_%d -j 1
Traceback (most recent call last):
  File "/home/user/.local/lib/python3.8/site-packages/grammarinator/generate.py", line 129, in create_new_test
    tree = generator(self.rule, self.max_depth)
  File "/home/user/.local/lib/python3.8/site-packages/grammarinator/generate.py", line 167, in generate
    unlexer = self.unlexer_cls(**dict(self.unlexer_kwargs, max_depth=max_depth))
  File "/home/user/userspace/php-gramm/out/PhpUnlexer.py", line 47, in __init__
    super(PhpUnlexer, self).__init__() 
TypeError: __init__() missing 1 required positional argument: 'input'
```

I tried to manually fix this, by figuring out what the hierarchy is of grammars-v4 `PhpLexerBase` which extends `Lexer` class. but it leads to other non-defined fields in `PhpUnlexer.py`. Seems, this integration is not working, or I'm missing something.

Please assist
Sabr

**Comment by renatahodovan** [comment]
Hi @Novitoll!

Thanks for your feedback! From the log, I guess that you are using the 19.3 release of Grammarinator (please correct me if I'm wrong).

The grammars of the grammars-v4 repository are primary created for parsing purposes. Even if we use them for generation there are some parts that don't fit into our use case, like inline code snippets or the original super classes of the lexers and parsers. The reason is simple, they encode additional instructions for the generated parser and not for the generator. To avoid these issues, you need to comment out the `superClass=PhpLexerBase` line in PhpLexer.g4 and you have to add the `--no-actions` cli argument to `grammarinator-process`. After these changes, PHP generation should work.

Note: you can use the superClass option of the ANTLR language by generation but its value should refer to a class that was written for the generator. Similarly, inline snippets are also supported by the generator but it should contain snippets targeting generation purposes.

I hope this helps, if you have further issues please don't hesitate to ask.

**Comment by renatahodovan** [comment]
@CityOfLight77 Thanks for the feedback. Please try to update Grammarinator to the latest master and regenerate your PhpGenerator. The issue was that the Php parser grammar contained a rule named `chain` - and hence PhpGenerator had a method with the same name - which collided with the `chain` method imported from `itertools`. This collision should be fixed in the latest master ( #38 ).

**Comment by renatahodovan** [comment]
@Novitoll @CityOfLight77 It seems that the suggested changes in the input grammar solved the first problem. While the second problem is fixed now by #38 . If you still experience any of these problems (or others) please don't hesitate to reopen this or open a new ticket.
